modules/importer/src/server/csaf/storage.rs

```diff
@@ -38,19 +36,17 @@ impl ValidatedVisitor for StorageVisitor {
         _context: &Self::Context,
         result: Result<ValidatedAdvisory, ValidationError>,
     ) -> Result<(), Self::Error> {
-        self.store(&result?.retrieved).await?;
-        Ok(())
-    }
-}
-
-impl StorageVisitor {
-    async fn store(&self, doc: &RetrievedAdvisory) -> Result<(), StorageError> {
-        let loader = CsafLoader::new(&self.system);
-
-        loader
-            .load(doc.url.as_str(), BufReader::new(doc.data.as_ref()))
+        let doc = result?;
+        let location = doc.context.url().to_string();
+
+        self.ingestor
+            .ingest(
+                &location,
+                Format::CSAF,
```

Comment:
Typing while I read, this may be answered below but...

Can we link the Format::CSAF/OSV etc to the actual path param in the REST API, instead of being stringly-typed as I left it?

Reply:
I assume this question will be answered by #122 ?

Reply:
Ahh. Kk!